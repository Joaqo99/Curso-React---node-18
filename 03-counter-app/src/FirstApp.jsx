import PropTypes from "prop-types";

const FirstApp = ({ title, subTitle, name })=>{
    return (
        <>
            <h1 data-testid="test-title">{ title }</h1>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>
    )
}

FirstApp.protoTypes = {
    title: PropTypes.string.isRequired
}

export default FirstApp

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    name: 'Joaco',
    subTitle: 'No hay subtitulo',
    title: 'No hay titulo'
}