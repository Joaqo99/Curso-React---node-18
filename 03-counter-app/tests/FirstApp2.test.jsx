import { render, screen } from "@testing-library/react"
import FirstApp from "../src/FirstApp"

describe('Pruebas en <FirstApp />', ()=>{
    const title = 'Hola, soy Joaquín'
    const subTitle = 'Soy un subtitulo'

    test('debe de hacer match con el snapshot',()=>{
        const { container } = render( <FirstApp title={title}/> )
        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar el mensaje "Hola, soy Joaquín"', () => {
        render( <FirstApp title={title}/> )
        expect( screen.getByText(title)).toBeTruthy()
    })

    test('debe de mostrar el titulo en h1',()=>{
        render( <FirstApp title={title} /> )
        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain( title )
    })


})