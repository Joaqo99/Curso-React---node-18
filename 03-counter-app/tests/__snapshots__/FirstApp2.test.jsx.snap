// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Joaqu√≠n
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    Joaco
  </p>
</div>
`;
